packages/cli/src/ui/hooks/useFooterProps.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { useUIState } from '../contexts/UIStateContext.js';
+import { useConfig } from '../contexts/ConfigContext.js';
+import { useSettings } from '../contexts/SettingsContext.js';
+
+export const useFooterProps = () => {
```

Comment:
we now have two basically identical blocks of logic computing the footer props. This one and the one for the non-screen reader layout. To avoid bugs, can we simplify. I'd be open to making  <Footer stop taking any props and just call this logic within the  definition of Footer.
Also ok with calling useFooterProps in the other case where we build up this list of properties so we don't have two duplicate versions that might diverge if we aren't careful as seen by the case here where vimMode isn't set correctly.